### DISCUSSION TITLE
Can't get the @parcel/optimizer-data-url to convert the images in url for backgrounds into inline base64

### DISCUSSION THREAD

**Discussion by hamzamihaidanielx** [parcel-bundler/parcel#5550] [Q&A] [answered]
As i was struggling with this, came across to this issue https://github.com/parcel-bundler/parcel/issues/4872, but can't quite get it how, nor in the docs say anything about this tho.

I mean, can't even get it how to resolve this `url(./../../assets/images/layout/footer.png) center no-repeat` without using the `import ... url:` thing. Also, previously, in v1, it was possible something like `url(~assets/images/layout/footer.png) center no-repeat` with ~ alias and got resolved.

Is there any way?

Thank you kindly!

**Comment by mischnic** [comment, marked as answer]
Something like `url(./../../assets/images/layout/footer.png)` should still work with Parcel 2. 

The only difference regarding `~` is that it points to a different folder in Parcel 2 in some cases: https://v2.parceljs.org/features/module-resolution/#tilde-paths

**Comment by hamzamihaidanielx** [comment]
So i got the following code:

[https://pastebin.com/qPE0BzWU](https://pastebin.com/qPE0BzWU)

Basically, If I were to try to use directly the imported Footer Image as `url(${FooterImage})` i would get the URL image but i would like it to have it directly inline as base64 and also the second image to use it something like `url(./../../assets/images....)`

**Reply by mischnic** [reply]
So using `data-url:` instead of `url:` doesn't work?

https://v2.parceljs.org/configuration/plugin-configuration/#predefined-(offical)-named-pipelines

**Reply by hamzamihaidanielx** [reply]
@mischnic Nope, it does not at all... :(

**Reply by hamzamihaidanielx** [reply]
Maybe due to the fact to the emotion css in js library?

**Reply by hamzamihaidanielx** [reply]
Doesn't it seems its gonna work with...css background, its weird

**Reply by mischnic** [reply]
What does it do instead? `FooterImage` should then be a string like `data:image/png;base64,iVB....`. It doesn't really matter where that is then used. Try console.log-ing that value.

The webp conversion probably doesn't happen though...

**Reply by hamzamihaidanielx** [reply]
Jeez...I'm such a retard sometimes 😂. 
Did this `import FooterImage from "data-url:./../../assets/images/layout/footer.png?as=webp&quality=1"` and then in the inline emote css in js prop css i added the following:

`background: url(${FooterImage}) center no-repeat`

And seems to work perfectly now, and its inline and the footer.webp doesn't get requested anymore in the network either.

Thank you!

**Reply by hamzamihaidanielx** [reply]
What's quite interesting is that the conversion doesn't take place anymore in data-url into webp format, this is an expected behavior right?

**Reply by mischnic** [reply]
Yeah, it's only enabled for `url:` at the moment:
https://github.com/parcel-bundler/parcel/blob/67aefd83f8a32813aa86c904952d6ceb63eb235c/packages/configs/default/index.json#L37

**Comment by hamzamihaidanielx** [comment]
Actually, doesn't it seems like I would need to write a custom plugin, in this case an optimizer or a transformer in order to parse and resolve the css in js URLs and get them converted to base64 and update the asset?
